deploy-board/deploy_board/webapp/cluster_view.py

```diff
@@ -220,6 +220,17 @@ def post(self, request, name, stage):
             cluster_name = env.get('clusterName')
             cluster_info = json.loads(request.body)
             log.info("Update Cluster Configuration with {}", cluster_info)
+            cluster_name = '{}-{}'.format(name, stage)
+            current_cluster = clusters_helper.get_cluster(request, cluster_name)
+            log.info("getting current Cluster Configuration is {}", current_cluster)
+            if 'spiffe_id' not in current_cluster and 'spiffe_id' in cluster_info:
+                log.info("Do not allow pass in spiffe_id if current cluster does not have one{}", cluster_info['spiffe_id'])
+                del cluster_info['spiffe_id']
+            if 'spiffe_id' in current_cluster and 'spiffe_id' in cluster_info:
+                if current_cluster['spiffe_id'] != cluster_info['spiffe_id']:
+                    log.info("Do not allow users to update customized spiffe_id {} when current spiffe_id {}",
+                    cluster_info['spiffe_id'], current_cluster['spiffe_id'])
+                    cluster_info['spiffe_id'] = current_cluster['spiffe_id']
```

Comment:
can we throw exception when the spiffe_id does not match? instead of silently ignore it.

Reply:
Do you mean if user pass in a different spiffe_id , we need throw exception and prevent user from calling update_cluster?
My code currently will upgrade user's spiffe_id for them, and continue.

Reply:
> can we throw exception when the spiffe_id does not match? instead of silently ignore it.
You mean when user pass in a different spiffe_id , we need throw an exception and prevent user from calling update_cluster?
My current code is upgrading user's spiffe_id, and continue call upgrade_cluster.
Is my current logic expected?

Reply:
Updated, please review.